### PR TITLE
docs: context for resources config

### PR DESCRIPTION
**Pull request by PaulRMellor**
Signed-off-by: prmellor <pmellor@redhat.com>

**Documentation**
Updates the `resources` common config section in the _Using_ guide.
Provides more context for configuration of `memory` and `cpu` requests and limits.
Adds diagram to illustrate resource request and limit

### Checklist

_Please go through this checklist and make sure all applicable tasks have been done_

- [ ] Write tests
- [ ] Make sure all tests pass
- [x] Update documentation
- [ ] Check RBAC rights for Kubernetes / OpenShift roles
- [ ] Try your changes from Pod inside your Kubernetes and OpenShift cluster, not just locally
- [ ] Reference relevant issue(s) and close them after merging
- [ ] Update CHANGELOG.md
- [ ] Supply screenshots for visual changes, such as Grafana dashboards

### CODE REVIEW

**Review by scholzj — COMMENTED** [review]
We are explaining many details here, but only small part is unique to Strimzi. I think we should do a better job linking to Kubernetes documentation explaining the basics and keep here only the Strimzi specific parts:
* Where to set the limits
* What are the recommended values (which we do not give as specific numbers, but as a list of things to consider and recommendations)

File: documentation/modules/con-common-configuration-properties.adoc [hunk 1]
```diff
@@ -101,70 +101,108 @@ For information on configuring TLS client authentication, see xref:type-KafkaCli
 [id='con-common-configuration-resources-{context}']
 === `resources`
 
-You request CPU and memory resources for components.
-Limits specify the maximum resources that can be consumed by a given container.
+Configure resource _requests_ and _limits_ to control resources for Strimzi containers.
+By reserving enough `memory` and `cpu` resources you can ensure a stable performance of Kafka.
```
[the remark this answers is not in the document]
Comment (PaulRMellor) [reply documentation/modules/con-common-configuration-properties.adoc:105]: Changed to:

> Configure resource _requests_ and _limits_ to control resources for Strimzi containers.
> By **requesting** enough `memory` and `cpu` resources you can ensure a stable performance of Kafka.

File: documentation/modules/con-common-configuration-properties.adoc [hunk 2]
```diff
@@ -101,70 +101,108 @@ For information on configuring TLS client authentication, see xref:type-KafkaCli
...
-* `cpu`
-* `memory`
+For Kafka, the following aspects of a deployment can impact the use of resources:
 
-Strimzi uses the Kubernetes syntax for specifying these resources.
+* Throughput and size of messages
+* The number of network threads handling messages
+* The number of producers and consumers
+* The number of topics and partitions
+* Storage volumes
```
[the remark this answers is not in the document]
Comment (PaulRMellor) [reply documentation/modules/con-common-configuration-properties.adoc:116]: You mention infrastructure (network and storage) at the end of the resources config demo on youtube. I can remove or update to something like this:

> For Kafka, the following aspects of a deployment can impact the resources you need:
> 
> * Throughput and size of messages
> * The number of network threads handling messages
> * The number of producers and consumers
> * The number of topics and partitions
> * Infrastructure, such as network or storage

[the remark this answers is not in the document]
Comment (scholzj) [reply documentation/modules/con-common-configuration-properties.adoc:116]: I'm not sure what exactly you refer to TBH. For sure storage or network can impact the performance. But you talk here about configuring the Kubernetes CPU and memory resources and I'm not sure that has any clear relationship to storage or networking. By giving it more CPU or memory, you do not improve slow storage or network.

File: documentation/modules/con-common-configuration-properties.adoc [hunk 3]
```diff
@@ -101,70 +101,108 @@ For information on configuring TLS client authentication, see xref:type-KafkaCli
...
+A container can use the resources up to the limit only when they are available.
+Resource limits are temporary and can be reallocated.
 
-Requests specify the resources to reserve for a given container.
-Reserving the resources ensures that they are always available.
+.Resource request and limit
+image:common-config-resources.png[Boundaries of a resource request and limit]
 
-IMPORTANT: If the resource request is for more than the available free resources in the Kubernetes cluster, the pod is not scheduled.
+If you don't set a limit, the container is given a default limit.
```
[the remark this answers is not in the document]
Comment (PaulRMellor) [reply documentation/modules/con-common-configuration-properties.adoc:127]: I removed this sentence. We explain this in the next sentence.

File: documentation/modules/con-common-configuration-properties.adoc (hunk 2 above)
[the remark this answers is not in the document]
Comment (PaulRMellor) [reply documentation/modules/con-common-configuration-properties.adoc]: I've removed. I guess we don't need introduce extraneous factors here.

**Comment by PaulRMellor** [comment]
This issue fixes #916

**Comment by PaulRMellor** [comment]
@scholzj - Hi Jakub. Do we have the level of detail you're expecting now?

**Review by scholzj — APPROVED** [review]

**Pull request merged**